prow/cmd/deck/github_oauth_setup.md

```diff
@@ -0,0 +1,119 @@
+# How to setup GitHub Oauth
+This document helps configure GitHub Oauth, which is required for [PR Status](https://prow.k8s.io/pr)
+and for the rerun button on [Prow Status](https://prow.k8s.io). 
+If OAuth is configured, Prow will perform GitHub actions on behalf of the authenticated users.
+This is necessary to fetch information about pull requests for the PR Status page and to 
+authenticate users when checking if they have permission to rerun jobs via the rerun button on Prow Status.
+
+## Set up secrets
+The following steps will show you how to set up an OAuth app.
+1. Create your GitHub Oauth application 
+
+    https://developer.github.com/apps/building-oauth-apps/creating-an-oauth-app/
+    
+    Make sure to create a GitHub Oauth App and not a regular GitHub App.
+    
+    The callback url should be:
+    
+    `<PROW_BASE_URL>/github-login/redirect`
+2. Create a secret file for GitHub OAuth that has the following content. The information can be found in the [GitHub OAuth developer settings](https://github.com/settings/developers):
+    
+    ```
+    client_id: <APP_CLIENT_ID>
```

Comment:
Doesn't this need to be JSON?

Reply:
Seems like it's YAML: https://github.com/kubernetes/test-infra/blob/4c871f02efffa9a6cd424644c0b0d1e8abe29907/prow/cmd/deck/main.go#L521